data/data_request.py

```diff
@@ -0,0 +1,162 @@
+import requests
+from datetime import datetime
+import pandas as pd
+
+urls = {
+    "coingecko": "https://api.coingecko.com/api/v3/coins/",
+}
+
+#TODO: Give each token a quote currency already and request prices and returns in this class.
+# In the Token_Pair class, check if both token habe the same quote currency, else fail.
+# If they do have the same quote currency, pair returns can be computed as ((1+r(A)) * (1+r(B)))
+class Token():
+    """Class that represents a token"""
+    def __init__(self, name:str, ticker: str):
+        """Initializing the token object.
+
+        Args:
+            name (str): Name of the token which is used to identify the token in the data query so this must match the name of the token for the specific source.
+                NOTE: The name can be different when it should be the quote currency (e.g. 'bitcoin' becomes 'btc' if it should be the quote currency instead of the base currency)
+            ticker (str): An abbriviation of the name between 3-5 letters&digits that will represent the pair in a quote (e.g. BTC/DOT)
+        """
+        self._ticker = ticker
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def ticker(self) -> str:
+        return self._ticker
+
+
+# TODO: Add a default token argument that sets the quote currency to USD if nothing else is specified.
+class Token_Pair():
+    """Class representating a trading pair of two tokens."""
+    def __init__(self, base_token: Token, quote_token: Token) -> None:
+        """Initializing the token pair.
+
+        Args:
+            base_token (Token): A token that represent the base token of the trading pair. 
+                This can be seen as '1 unit of base token is worth x units of quote token'
+            quote_token (Token): A second token that represents the quote token of the pair. 
+                This can be seen as 'x unit of quote token is worth 1 unit of base token'
+        """
+        self._base_token = base_token
+        self._quote_token = quote_token
+
+    # Getter & Setter
+
+    @property
+    def base_token(self) -> Token:
+        return self._base_token
+
+    @property
+    def quote_token(self) -> Token:
+        return self._quote_token
+
+    @property
+    def prices(self) -> pd.DataFrame:
+        return self._prices
+    
+    # TODO: Do I even need setter if I only set the values inside the class? I don't think so!
```

Comment:
The setter would make sense if you plan on doing some processing of the `prices` parameter before actually storing. If you intend to keep this function, you should use this setter in the other methods instead of e.g. `self.prices = 1 / prices`.

As a side note, `self._prices` could be initialized with `None` in the constructor to avoid uninitialized usage.

Reply:
My understanding was that with this getter&setter structure you want to prevent users or code to change these values from outside the class. I thought the main reason to have setters was to be able to set values from outside the class e.g `BTCaUSD.prices = [23000,22000....]. However, I think in this case there is no reason why someone would want to change prices or returns from outside the class so I think I can remove those setters, right?